Author: Mort65, https://github.com/mort65/
Description: This patch limits the length of WM_NAME on the status bar.

diff -up ../dwm-6.1/config.def.h ./config.def.h
--- ../dwm-6.1/config.def.h	2015-11-09 02:09:37.000000000 +0330
+++ ./config.def.h	2019-01-07 13:00:20.831711037 +0330
@@ -13,6 +13,7 @@ static const char selbgcolor[]      = "#
 static const char selfgcolor[]      = "#eeeeee";
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
+static const unsigned int maxwnln   = 64;       /* maximum length of windows names on the status bar */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */

diff -up ../dwm-6.1/dwm.c ./dwm.c
--- ../dwm-6.1/dwm.c	2015-11-09 02:09:37.000000000 +0330
+++ ./dwm.c	2019-01-07 12:58:35.788379009 +0330
@@ -742,6 +742,9 @@ drawbar(Monitor *m)
 	if ((w = x - xx) > bh) {
 		x = xx;
 		if (m->sel) {
+			if (strlen(m->sel->name) > maxwnln) {
+				m->sel->name[maxwnln] = '\0';
+			}
 			drw_setscheme(drw, m == selmon ? &scheme[SchemeSel] : &scheme[SchemeNorm]);
 			drw_text(drw, x, 0, w, bh, m->sel->name, 0);
 			drw_rect(drw, x + 1, 1, dx, dx, m->sel->isfixed, m->sel->isfloating, 0);
